src/ethereum/utils.py

```diff
@@ -59,3 +76,354 @@ def ceil32(value: Uint) -> Uint:
         return value
     else:
         return value + ceiling - remainder
+
+
+def to_valid_address(data: bytes) -> Address:
+    """
+    Convert a byte stream to a valid address (20 bytes).
+
+    Parameters
+    ----------
+    data :
+        The string to be converted to bytes.
+
+    Returns
+    -------
+    byte_stream : `bytes`
+        The byte stream obtained from the passed string.
+    """
+    if len(data) == 20:
+        return data
+
+    if len(data) < 20:
+        return data.rjust(20, b"\x00")
+
+    return data[-20:]
+
+
+def to_bytes(data: str) -> Bytes:
+    """
+    Convert a string to bytes.
+
+    Parameters
+    ----------
+    data :
+        The string to be converted to bytes.
+
+    Returns
+    -------
+    byte_stream : `bytes`
+        The byte stream obtained from the passed string.
+    """
+    if data is None:
+        return b""
+    if has_hex_prefix(data):
+        return hex_to_bytes(data)
+
+    return data.encode()
+
+
+def has_hex_prefix(hex_string: str) -> bool:
+    """
+    Check if a hex string starts with hex prefix (0x).
+
+    Parameters
+    ----------
+    hex_string :
+        The hexadecimal string to be checked for presence of prefix.
+
+    Returns
+    -------
+    has_prefix : `bool`
+        Boolean indicating whether the hex string has 0x prefix.
+    """
+    return hex_string.startswith("0x")
+
+
+def remove_hex_prefix(hex_string: str) -> str:
+    """
+    Remove 0x prefix from a hex string if present. This function returns the
+    passed hex string if it isn't prefixed with 0x.
+
+    Parameters
+    ----------
+    hex_string :
+        The hexadecimal string whose prefix is to be removed.
+
+    Returns
+    -------
+    modified_hex_string : `str`
+        The hexadecimal string with the 0x prefix removed if present.
+    """
+    if has_hex_prefix(hex_string):
+        return hex_string[len("0x") :]
+
+    return hex_string
```

Comment:
This is probably a huge preference thing, but I would much rather we _require_ a hex prefix, rather than allowing it.

Thoughts?

Reply:
I don't know, this utility deals well with both non-prefixed and prefixed hex strings. So it might be a good to have feature right?

Reply:
I'm not sure. If you require the `0x` prefix, it's unambiguous that the data is hex.

Like I said, not a huge deal though.

Reply:
I would prefer we require prefixed hex strings, but there are a handful of places in eth tooling where naked hex strings are used. :/

Reply:
I'm too lazy to dig up my Emperor Palpatine "I am the senate" meme, but I'm sure you'll remember it ;)

Reply:
haha